File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        right_pointer = len(numbers) - 1
        left_pointer = 0

        while (left_pointer < right_pointer ):
            sum = numbers[left_pointer] + numbers[right_pointer]
            if (sum == target):
                return [left_pointer+1, right_pointer+1]
            elif (sum > target):
                right_pointer -=1
            else :
                left_pointer +=1 

        return [-1,-1];        
        

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n = numbers.length;
        int leftPointer = 0;
        int rightPointer = n-1;

        while (leftPointer < rightPointer ){
            int sum = numbers[leftPointer] + numbers[rightPointer];
            if (sum == target){
                return new int[]{leftPointer + 1, rightPointer + 1};
            } else if (sum < target){
                leftPointer++;
            } else {
                rightPointer--;
            }

        }
        return new int[]{-1,-1};
    }
}